src/Microsoft.DotNet.XHarness.WebAssembly/xunit-runner.cs

```diff
@@ -0,0 +1,395 @@
+                    nfail++;
+                    continue;
+                }
+
+                if (tc is Xunit.Sdk.XunitTheoryTestCase)
+                {
+                    if (method.IsGenericMethod)
+                    {
+                        Console.WriteLine("SKIP (generic): " + tc.DisplayName);
+                        nfiltered++;
+                        continue;
+                    }
+
+                    // From XunitTheoryTestCaseRunner
+                    IEnumerable<IAttributeInfo> attrs = tc.TestMethod.Method.GetCustomAttributes(typeof(DataAttribute));
+                    bool failed = false;
+                    foreach (IAttributeInfo dataAttribute in attrs)
+                    {
+                        IAttributeInfo discovererAttribute = dataAttribute.GetCustomAttributes(typeof(DataDiscovererAttribute)).First();
+                        var args = discovererAttribute.GetConstructorArguments().Cast<string>().ToList();
+
+                        Type discovererType = null;
+                        if (args[1] == "xunit.core")
+                            discovererType = typeof(IXunitTestCollectionFactory).Assembly.GetType(args[0]);
+                        if (discovererType == null)
+                        {
+                            Console.WriteLine("FAIL (discoverer): " + args[0] + " " + args[1]);
+                            failed = true;
+                        }
+
+                        IDataDiscoverer discoverer;
+                        discoverer = ExtensibilityPointFactory.GetDataDiscoverer(this, discovererType);
+
+                        try
+                        {
+                            IEnumerable<object[]> data = discoverer.GetData(dataAttribute, tc.TestMethod.Method);
+                            object[][] data_arr = data.ToArray();
+                            Console.WriteLine(tc.DisplayName + " [" + data_arr.Length + "]");
+                            foreach (object[] dataRow in data_arr)
+                            {
+                                nrun++;
+                                object obj = null;
+                                if (!method.IsStatic)
+                                {
+                                    ConstructorInfo constructor = method.ReflectedType.GetConstructor(Type.EmptyTypes);
+                                    if (constructor != null)
+                                        obj = constructor.Invoke(null);
+                                    else
+                                        obj = System.Runtime.Serialization.FormatterServices.GetUninitializedObject(method.ReflectedType);
+                                }
+
+                                ParameterInfo[] pars = method.GetParameters();
+                                for (int i = 0; i < dataRow.Length; ++i)
+                                    dataRow[i] = ConvertArg(dataRow[i], pars[i].ParameterType);
```

Comment:
Looking at https://github.com/dotnet/xharness/pull/171/files#diff-58c3c1225cc3c789afd92878aa749711R157 if that method catches an exception, the returned arg is not going to be converted. Is that what we want to be doing. Will mean that it will be the same object that was input.

Reply:
Is this a problem?

Reply:
I am puzzle because, if there is no need for the conversion, then we don't need the method at all, right?